app/javascript/src/ui/hooks/useCallTransaction.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useCallback } from "react";
+
+import { NormalizedTransaction, normalizeTransaction } from "api/utils/normalizeTransaction";
+import { TransactionState } from "ui/hooks/types";
+
+export interface CallTransactionState<T> extends TransactionState<T> {
+  called: boolean;
+  call: (...args: any[]) => Promise<NormalizedTransaction<T>>;
+}
+
+export const useCallTransaction = <T>(transaction: Function): CallTransactionState<T> => {
```

Comment:
`useCallTransaction` is equivalent to `registerAction` - it returns a function that can be used to manually initiate a transaction. But within calling it directly, the transaction will not occur.

`useAutonomousTransaction` is equivalent to `registerDependency` - it syncs the transaction to component lifecycle. The transaction will be called when the component mounts and then any time that the transaction arguments change. It also exposes a function that allows consumers to manually trigger it.

I considered combining these into one hook, but liked the separation between the two. I'm also not a huge fan of the name `useAutonomousTransaction`, but couldn't think of anything better at the time.

Reply:
If the analogy holds, could we name them `useReadTransaction` and `useWriteTransaction`? Thats how `registerDependency` and `registerAction` roughly map anyways with regard to API transactions

Reply:
Also, just to clarify, this doesn't do anything with redux right? `registerDependency` and `registerAction` actually tie into the redux store. As do their associated hooks `useQuery` and `useMutation`

I just wanna make sure that when we start using hooks in places, we don't build sneaky bugs because that data isn't in the store.  That said, a lot of places don't need redux and could just use state instead.